### ISSUE TITLE
Find out which data were used to train the model

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/deepseg_lesion_models#2]
@charleygros I hope all is well with you 😊 

Could you please help us find out which data were used to train/validate/test the `sct_deepseg_lesion` model? Is there some sort of config file or scripts that lists the dataset?

Thank you!

**Comment by charleygros** [comment]
@jcohenadad all is well, thank you!
I also hope things are going well for you :-)

Just a message to say that I've received this notification: I will have a first look for it this weekend and come back to you.
Speak soon

**Comment by charleygros** [comment]
First look at my hard drive: nothing here.
But I have good hope it is on my old laptop: which I won't get to until late next week, hope that's ok.
Speak soon.

**Comment by jcohenadad** [comment]
Would it help if we go through our `duke` server together? Digging a bit, I found this for example:

![Screen Shot 2023-06-24 at 10 30 29 AM](https://github.com/spinalcordtoolbox/deepseg_lesion_models/assets/2482071/97e8d79a-89c1-4df7-8004-9ee6f95637e4)

Is this relevant?

**Comment by charleygros** [comment]
Hello, I came back to my old laptop and check everything there: nothing, sadly.

I think `duke` is relevant for sure!

Would it be possible to give me access to this `ms_seg` folder? Or to upload it to a dropbox if not too heavy?
I'm keen to try to dig this out. I have some time for this next week.

Also... I vaguely remember that when we uploaded the models to SCT, we had a `json` or a `txt` file in the same folder that contained the filenames used for training.
(or maybe was it for the lesion map.. mmmh)
Looking this way now.

**Comment by charleygros** [comment]
> Also... I vaguely remember that when we uploaded the models to SCT, we had a `json` or a `txt` file in the same folder that contained the filenames used for training.
> (or maybe was it for the lesion map.. mmmh)

No luck here.

`duke` is probably the next promising place to look.
```
ms_seg
    /sc_auto_seg
    /sct_deepseg_sc_info
    /seg_paper
```

@jcohenadad : I'm happy to look into these folders if this helps

**Comment by jcohenadad** [comment]
> @jcohenadad : I'm happy to look into these folders if this helps

@charleygros thank you so much! Yes, that would be great. @naga-karthik can you please get in touch with Charley so you two can navigate duke?

**Comment by naga-karthik** [comment]
Absolutely! @charleygros, just sent you an email for scheduling a meeting!

**Comment by naga-karthik** [comment]
Thanks to @charleygros's accurate memory, we were able to find the data used for `deepseg_lesion` pretty quickly! Here are some notes for reference: 

* The MS data can be found under: `duke/projects/ms_seg/seg_paper/data_ms`. Only data of MS patients (not healthy controls, HCs) were used for training and validation but the model was tested on both MS and HCs. 
* There's a [dataset.pkl](https://github.com/spinalcordtoolbox/deepseg_sc_models/blob/master/src/spinalcordtoolbox/data/deepseg/sc/dataset.pkl) containing the information on the subjects used, their pathology, train/val/test, etc that was used for `deepseg_sc`. From this, we have to pick out only MS patients (specified under the column `patho`) and split them according to the train/val/test split. This should give us an exact comparison between the `bavaria-quebec` model tested on `deepseg_lesion` model's test data and `deepseg_lesion` model's test results. 
* The other folders - `/sc_auto_seg` and `/sct_deepseg_sc_info` are not relevant as they were only used for experimental stuff.

**Comment by jcohenadad** [comment]
Amazing! Thank you @charleygros!

@naga-karthik can you please add the relevant/necessary info in the README? thanks!

**Comment by naga-karthik** [comment]
Done! @jcohenadad seems like the main branch is protected. It would be great if you could merge [my PR](https://github.com/spinalcordtoolbox/deepseg_lesion_models/pull/3) that adds this info to the readme! Thank you!

**Comment by jcohenadad** [comment]
Fixed by #3
